src/routes/news/containers/NewsPage.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import * as React from 'react'
+import { connect } from 'react-redux'
+import compose from 'lodash/fp/compose'
+import { LocalNewsModel } from '@integreat-app/integreat-api-client'
+import ContentNotFoundError from '../../../modules/common/errors/ContentNotFoundError'
+import FailureSwitcher from '../../../modules/common/components/FailureSwitcher'
+import type { TFunction } from 'react-i18next'
+import { withTranslation } from 'react-i18next'
+import type { StateType } from '../../../modules/app/StateType'
+import NewsElement from '../components/NewsElement'
+import NewsList from '../components/NewsList'
+import Tabs from '../components/Tabs'
+import PushNewsDetails from '../components/PushNewsDetails'
+
+type PropsType = {|
+  news: Array<LocalNewsModel>,
+  city: string,
+  newsId: ?string,
+  language: string,
+  t: TFunction,
+  path: string
+|}
+
+const LOCAL_NEWS = 'local'
+const TU_NEWS = 'tu'
+
+/**
+ * Displays a list of news or a single newsElement, matching the route /<location>/<language>/news(/<id>)
+ */
+export class NewsPage extends React.Component<PropsType> {
+  state = {
+    hasNextPage: true,
+    isNextPageLoading: false,
+    items: []
+  };
+
+  renderNewsElement = (language: string, type: string) => (newsItem: EventModel) => (
+    <NewsElement newsItem={newsItem} language={language} key={newsItem.path} type={type} path={this.props.path} />
+  )
+
+  render () {
+    const { news, path, newsId, city, language, t } = this.props
+
+    if (newsId) {
```

Comment:
I suggest if we move details component to separate route will be better, maybe different page so we keep render component neat an clean and make newsPage Component to show list only

What do you think?

Reply:
Sounds good. This is required for the React-Native app aswell in order to allow navigating back.

Reply:
I think we should leave it like this to be consistent with rest of existing routes.

Reply:
Moving the details to separate routes sounds good. The reason why we don't do this for most of the other routes is that we don't use single endpoints there but instead just load everything with one request. An example for separation of list and detail components are the `extras` route and its detail routes `sprungbrett` and `wohnen`.